Simplify 'requestPaintingResource' function

Ahmad-S792 opened the pull request:
#### 56453b537abfa43027140bca7229499250d4ebc7
<pre>
Simplify &apos;requestPaintingResource&apos; function

<a href="https://bugs.webkit.org/show_bug.cgi?id=252027">https://bugs.webkit.org/show_bug.cgi?id=252027</a>

Reviewed by Said Abou-Hallawa.

Inspired - <a href="https://chromium.googlesource.com/chromium/blink/+/54571499b970b85114c1ecb39cbde9bdbbf61d07">https://chromium.googlesource.com/chromium/blink/+/54571499b970b85114c1ecb39cbde9bdbbf61d07</a>

This patch simplifies the &quot;requestPaintingResource&apos; function and removes extra code / calls.

* Source/WebCore/rendering/svg/RenderSVGResource.cpp:
(requestPaintingResource): As above in commit message

Canonical link: <a href="https://commits.webkit.org/260686@main">https://commits.webkit.org/260686@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/902c4afbcdde6a6a2c0c9b7f4220b76c29306d47

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/109072 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/18154 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/41894 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/607 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/118338 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/112957 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/19613 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/9456 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/101313 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/114832 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/14701 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/97933 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/42841 "Failed to checkout and rebase branch from PR 9966") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/96673 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/29586 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/84592 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/10936 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/30932 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/11674 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/7863 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/17048 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/50531 "Passed tests") | | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7372 "Failed to push commit to Webkit repository") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/13281 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

shallawa commented on Source/WebCore/rendering/svg/RenderSVGResource.cpp:
```diff
@@ -76,8 +78,7 @@ static inline RenderSVGResource* requestPaintingResource(RenderSVGResourceMode m
     SVGPaintType paintType = applyToFill ? svgStyle.fillPaintType() : svgStyle.strokePaintType();
-    if (paintType == SVGPaintType::None)
-        return nullptr;
+    ASSERT(paintType != SVGPaintType::None);
```
I think the part of this function from the beginning till this line can be simplified like this.

```
    bool applyToFill = mode == RenderSVGResourceMode::ApplyToFill;

    // When rendering the mask for a RenderSVGResourceClipper, ...
    if (renderer.view().frameView().paintBehavior().contains(PaintBehavior::RenderingSVGMask)) {
        // Ignore stroke.
        if (!applyToFill)
            return nullptr;

        // But always use the initial fill paint server.
        RenderSVGResourceSolidColor* colorResource = RenderSVGResource::sharedSolidPaintingResource();
        colorResource->setColor(SVGRenderStyle::initialFillPaintColor().absoluteColor());
        return colorResource;
    }

    const auto& svgStyle = style.svgStyle();
    auto paintType = applyToFill ? svgStyle.fillPaintType() : svgStyle.strokePaintType();

    // If we have no fill/stroke, return nullptr.
    if (paintType == SVGPaintType::None)
        return nullptr;
```

Ahmad-S792 replied:
Done and pushed commit. I think EWS hook will take some time to run it through EWS.

shallawa commented on Source/WebCore/rendering/svg/RenderSVGResource.cpp:
```diff
@@ -54,28 +55,24 @@ static inline bool inheritColorFromParentStyleIfNeeded(RenderElement& object, bo
-            return nullptr;
-    } else {
-        if (!svgStyle.hasStroke() || isRenderingMask)
+    // When rendering the mask for a RenderSVGResourceClipper, always use the initial fill paint server, and ignore stroke.
```
This comment is duplicated inside the if-statement. Please rephrase it, or delete the comments inside the if-statement.

Ahmad-S792 replied:
On it. :-)

Ahmad-S792 replied:
Rephrasing it to remove 'ignore stroke' part since it is repeating samething as you pointed out.

Ahmad-S792 replied:
Done, pushed new commit. Marking this as "Resolve" and will add "merge-queue".

Ahmad-S792 committed:
Simplify 'requestPaintingResource' function

https://bugs.webkit.org/show_bug.cgi?id=252027

Reviewed by Said Abou-Hallawa.

Inspired - https://chromium.googlesource.com/chromium/blink/+/54571499b970b85114c1ecb39cbde9bdbbf61d07

This patch simplifies the "requestPaintingResource' function and removes extra code / calls.

* Source/WebCore/rendering/svg/RenderSVGResource.cpp:
(requestPaintingResource): As above in commit message

Canonical link: https://commits.webkit.org/260686@main
